frontend/src/components/DetailsTable.tsx

```diff
@@ -48,22 +47,29 @@ export const css = stylesheet({
   },
 });
 
-interface DetailsTableProps {
-  fields: Array<KeyValue<string | S3Artifact>>;
+export interface ValueComponentProps<T> {
+  value?: string | T;
+  [key: string]: any;
+}
+
+interface DetailsTableProps<T> {
+  fields: Array<KeyValue<string | T>>;
   title?: string;
-  valueComponent?: React.FC<{ value: S3Artifact }>;
+  valueComponent?: React.FC<ValueComponentProps<T>>;
+  [key: string]: any;
```

Comment:
Thanks! this looks like a good approach.
nit: probably slightly change it to
```
valueComponentProps: { [key: string]: any };
```
to make it more focused.

Otherwise, DetailsTable will accept any key as a prop, which is not good for type checking.